src/flat-button.jsx

```diff
@@ -232,17 +199,23 @@ const FlatButton = React.createClass({
     } = this.props;
 
     const {
-      buttonColor,
-      buttonHeight,
-      buttonMinWidth,
-      disabledTextColor,
-      buttonFilterColor,
-      primaryTextColor,
-      secondaryTextColor,
-      textColor,
-      textTransform,
-    } = this.constructor.getRelevantContextKeys(this.state.muiTheme);
-
+      button: {
+        color: buttonColor,
+        height: buttonHeight,
+        minWidth: buttonMinWidth,
+        textTransform: buttonTextTransform,
```

Comment:
The logic is the same, `textTransform` is being assigned to a variable called `buttonTextTransform`, then in [line 216](https://github.com/callemall/material-ui/pull/3331/files#diff-25151f62042b3bd6779c91508a11cb56R216), if textTransform was not provided in flatButton theme, it defaults to either the buttonTextTransform or uppercase.